contracts/core/OTCWrapper.sol

```diff
@@ -0,0 +1,690 @@
+        address collateralAsset,
+        uint256 premium,
+        address indexed seller,
+        uint256 indexed vaultID,
+        address oToken,
+        uint256 initialMargin
+    );
+
+    /// @notice emits an event when collateral is deposited
+    event CollateralDeposited(uint256 indexed orderID, uint256 amount);
+
+    /// @notice emits an event when collateral is withdrawn
+    event CollateralWithdrawn(uint256 indexed orderID, uint256 amount);
+
+    /// @notice emits an event when a vault is settled
+    event VaultSettled(uint256 indexed orderID);
+
+    /************************************************
+     *  STORAGE
+     ***********************************************/
+
+    ///@notice order counter
+    uint256 public latestOrder;
+
+    ///@notice fill deadline duration
+    uint256 public fillDeadline;
+
+    // usdc address
+    address public USDC;
+
+    ///@notice address that will receive the product fees
+    address public beneficiary;
+
+    // order status
+    enum OrderStatus {
+        Pending,
+        Succeeded,
+        Failed
+    }
+
+    // struct defining order details
+    struct Order {
+        // underlying asset address
+        address underlying;
+        // collateral asset address
+        address collateral;
+        // option type the vault is selling
+        bool isPut;
+        // option strike price
+        uint256 strikePrice;
+        // option expiry timestamp
+        uint256 expiry;
+        // order premium amount
+        uint256 premium;
+        // order notional
+        uint256 notional;
+        // buyer address
+        address buyer;
+        // seller address
+        address seller;
+        // id of the vault
+        uint256 vaultID;
+        // otoken address
+        address oToken;
+        // timestamp of when the order was opened
+        uint256 openedAt;
+    }
+
+    // struct defining permit signature details
+    struct Permit {
+        // permit amount
+        uint256 amount;
+        // permit deadline
+        uint256 deadline;
+        // permit account
+        address acct;
+        // v component of permit signature
+        uint8 v;
+        // r component of permit signature
+        bytes32 r;
+        // s component of permit signature
+        bytes32 s;
+    }
+
+    // struct defining the upper and lower boundaries of notional for an asset
+    struct MinMaxNotional {
+        // minimum notional value allowed
+        uint256 min;
+        // maximum notional value allowed
+        uint256 max;
+    }
+
+    ///@dev mapping between an asset address to a struct consisting of uint256 min and a uint256 max which will be its notional floor and cap size allowed
+    mapping(address => MinMaxNotional) public minMaxNotional;
+
+    ///@notice mapping between order id and order details
+    mapping(uint256 => Order) public orders;
+
+    ///@notice mapping between order id and order status
+    mapping(uint256 => OrderStatus) public orderStatus;
+
```

Comment:
lets add one more mapping from user => array of succeeded orders

```
///@notice mapping between acct and list of all successful orders
mapping(address => uint[]) public ordersByAcct;
```

Reply:
https://github.com/ribbon-finance/GammaProtocol-OTC/pull/2/commits/6934947c3273adb1932b3b6251bea9e452471c41